### PR TITLE
test_schemaview.py: minor tidying, fixture creation

### PR DESCRIPTION
**Pull request by ialarmedalien**
Making parts of `test_schemaview.py` more consistent:
- importing/using some "constants" from `schemaview.py`
- create fixtures of some of the schemaviews to avoid recreating them for multiple tests
- load Path objects directly into SchemaView as it can handle them directly now

### CODE REVIEW

File: tests/test_utils/test_schemaview.py [hunk 1]
```diff
@@ -79,27 +93,51 @@ def schema_view_with_imports() -> SchemaView:
     return SchemaView(SCHEMA_WITH_IMPORTS)
 
 
+@pytest.fixture(scope="session")
+def sv_import_tree() -> SchemaView:
+    """Fixture for a SchemaView for testing imports and ordering."""
+    return SchemaView(INPUT_DIR_PATH / "imports" / "main.yaml")
```
Comment (ialarmedalien) [inline tests/test_utils/test_schemaview.py:99]: create a fixture instead of generating the SV in the test

File: tests/test_utils/test_schemaview.py [hunk 2]
```diff
@@ -79,27 +93,51 @@ def schema_view_with_imports() -> SchemaView:
...
+
+
 @pytest.fixture(scope="session")
 def sv_attributes() -> SchemaView:
     """Fixture for a SchemaView for testing attribute edge cases."""
-    return SchemaView(os.path.join(INPUT_DIR, "attribute_edge_cases.yaml"))
+    return SchemaView(INPUT_DIR_PATH / "attribute_edge_cases.yaml")
```
Comment (ialarmedalien) [inline tests/test_utils/test_schemaview.py:105]: no need for that ugly `os.path.join`

File: tests/test_utils/test_schemaview.py [hunk 3]
```diff
@@ -364,7 +402,7 @@ def test_imports(schema_view_with_imports: SchemaView) -> None:
     logger.debug(view.imports_closure())
     assert set(view.imports_closure()) == {"kitchen_sink", "core", "linkml:types"}
 
-    for t in view.all_types().keys():
+    for t in view.all_types():
```
Comment (ialarmedalien) [inline tests/test_utils/test_schemaview.py:405]: no need for `.keys()`

File: tests/test_utils/test_schemaview.py [hunk 4]
```diff
@@ -773,7 +809,9 @@ def test_uris_without_default_prefix() -> None:
         CREATURE_EXPECTED["element"][s_name].update(el)
 
 
-@pytest.mark.parametrize("schema", ["creature_view", "creature_view_remote", "creature_view_local"])
+@pytest.mark.parametrize(
+    "schema", ["creature_view", "creature_view_remote", "creature_view_local", "creature_view_direct_url"]
+)
```
Comment (ialarmedalien) [inline tests/test_utils/test_schemaview.py:814]: add in an extra test case for testing `imports`

File: tests/test_utils/test_schemaview.py [hunk 5]
```diff
@@ -1440,33 +1492,30 @@ def test_get_classes_by_slot(schema_view_with_imports: SchemaView) -> None:
     assert sorted(actual_result) == sorted(expected_result)
 
 
-def test_materialize_patterns() -> None:
+def test_materialize_patterns(sv_structured_patterns: SchemaView) -> None:
```
Comment (ialarmedalien) [inline tests/test_utils/test_schemaview.py:1495]: use the fixture instead of creating an SV each time

File: tests/test_utils/test_schemaview.py [hunk 6]
```diff
@@ -1585,44 +1633,44 @@ def test_merge_schema_no_clobber(sv_merge_1: SchemaView, sv_merge_2: SchemaView)
...
-    for element in ELEMENTS:
+    for element in ALL_ELEMENTS:
         (field, val) = _get_clobbered_field_val(element)
         for k, v in getattr(sv_merge_2.schema, element).items():
             if k in getattr(sv_merge_1.schema, element):
                 assert getattr(v, field) == val
 
 
-def test_get_slots_by_enum_schema_slot(sv_issue_998) -> None:
+def test_get_slots_by_enum_schema_slot(sv_issue_998: SchemaView) -> None:
```
Comment (ialarmedalien) [inline tests/test_utils/test_schemaview.py:1649]: add in typing

File: tests/test_utils/test_schemaview.py [hunk 7]
```diff
@@ -927,7 +966,20 @@ def test_alias_slot(schema_view_no_imports: SchemaView) -> None:
...
+    CAT:
+    LION:
+      is_a: CAT
+    ANGRY_LION:
+      is_a: LION
+    TABBY:
+      is_a: CAT
+    BIRD:
+    EAGLE:
+      is_a: BIRD
```
[the remark this answers is not in the document]
Comment (ialarmedalien) [reply tests/test_utils/test_schemaview.py:980]: It used to be in the schemaview.py module but I removed it in a recent PR and moved it here instead.
